sdk/keyvault/keyvault-admin/test/public/accessControlClient.spec.ts

```diff
@@ -0,0 +1,107 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { assert } from "chai";
+import { env, Recorder } from "@azure/test-utils-recorder";
+
+import { KeyVaultAccessControlClient } from "../../src";
+import { authenticate } from "../utils/authentication";
+
+describe("KeyVaultAccessControlClient", () => {
+  let client: KeyVaultAccessControlClient;
+  let recorder: Recorder;
+  let generateFakeUUID: () => string;
+  const globalScope = "/";
+
+  beforeEach(async function() {
+    const authentication = await authenticate(this);
+    client = authentication.accessControlClient;
+    recorder = authentication.recorder;
+    generateFakeUUID = authentication.generateFakeUUID;
+  });
+
+  afterEach(async function() {
+    await recorder.stop();
+  });
+
+  // The tests follow
+
+  it("listRoleDefinitions", async function() {
+    const expectedType = "Microsoft.Authorization/roleDefinitions";
+    let receivedRoles: string[] = [];
+
+    for await (const roleDefinition of client.listRoleDefinitions(globalScope)) {
+      // Each role definition will have the shape of:
+      //
+      //   {
+      //     id: 'Microsoft.KeyVault/providers/Microsoft.Authorization/roleDefinitions/<ID>',
+      //     name: '<ID>',
+      //     type: '<role-type>',
+      //     roleName: '<role-name>',
+      //     // ...
+      //   }
+      //
+      assert.equal(roleDefinition.type, expectedType);
+      receivedRoles.push(roleDefinition.roleName!);
+    }
+
+    // Roles might change
+    assert.ok(receivedRoles.length);
+  });
+
+  it("listRoleAssignments", async function() {
+    const expectedType = "Microsoft.Authorization/roleAssignments";
+    let receivedRoles: string[] = [];
+
+    for await (const roleAssignment of client.listRoleAssignments(globalScope)) {
+      // Each role assignment will have the shape of:
+      //
+      //   {
+      //     id: '/providers/Microsoft.Authorization/roleAssignments/<ID>',
+      //     name: '<ID>',
+      //     type: '<role-type>',
+      //     // ...
+      //   }
+      //
+      assert.equal(roleAssignment.type, expectedType);
+      receivedRoles.push(roleAssignment.name);
+    }
+
+    // Roles might change
+    assert.ok(receivedRoles.length);
+  });
+
+  it("createRoleAssignment, getRoleAssignment and deleteRoleAssignment", async function() {
+    // First, deleting any existing assignment, just in case.
+    for await (const roleAssignment of client.listRoleAssignments(globalScope)) {
```

Comment:
I believe it would be better to just create a new role assignment and then delete it, instead of deleting all existing role assignments for this principal.

Reply:
the issue is if the principal already has this role, how would you ensure that the principal doesn't have it before running the tests?